### PR TITLE
Remove SendSeries

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR removes `SendSeries` in favor of `SendIterableSeries`.
`IterableSeries` implements the interface `JSONMarshaler` which means `SendIterableSeries` can be used even when `enableJSONStream` is `false`.

Reviewing commit by commit is strongly recommended (some code was moved).

### Additional Notes

This PR is the same as https://github.com/DataDog/datadog-agent/pull/11133 except the following commits:
- de264649434344ec90e91225ec94c28475e3dad5: Note: `aggregator_test.go` was not updated
- fd7dcaeabcbc46697706f898cf040d8c1acd3fb0: Only two imports were updated.
- 3ade8a58a5cefe8dcdca633cdbed0e89278eb921: Resolve conflict with https://github.com/DataDog/datadog-agent/pull/11496.

### Describe how to test/QA your changes

Set `enable_stream_payload_serialization` at `false`, send `100 000` series and checks they are received by the web app (Agent core).
Check `serializerConsumer` works for series in OTLP (Agent platform).

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
```
=== Failed
=== FAIL: pkg/aggregator TestRecurrentSeries (unknown)
panic: 
assert: mock: The method has been called over 1 times.
	Either do one more Mock.On("SendServiceChecks").Return(...), or remove extra call.
	This call was unexpected:
		SendServiceChecks(metrics.ServiceChecks)
		0: metrics.ServiceChecks{(*metrics.ServiceCheck)(0xc0004b3560)}
	at: [test_common.go:30 aggregator.go:566 aggregator.go:597 aggregator.go:666 aggregator.go:687 asm_amd64.s:1581]

```

File: pkg/serializer/internal/metrics/iterable_series.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (series IterableSeries) MarshalJSON() ([]byte, error) {
 	// use an alias to avoid infinite recursion while serializing a Series
 	type SeriesAlias Series
 
-	var seriesAlias SeriesAlias
```
Comment (djmitche) [inline pkg/serializer/internal/metrics/iterable_series.go]: I've made this kind of mistake a few times -- maybe this instance is worth a quick unit test so it doesn't regress on the next refactor?

**Review by djmitche — APPROVED** [review]
I can confirm that each of these commits does what it says, and I love the amount of removed code!  But I don't have the context to be able to know this all fits together correctly.  In past reviews @vickenty caught some of those, so I think he should have a look as well.

File: pkg/aggregator/demultiplexer.go [hunk 2]
```diff
@@ -643,6 +643,7 @@ func sendIterableSeries(serializer serializer.MetricSerializer, start time.Time,
 	err := serializer.SendIterableSeries(series)
 	// if err == nil, SenderStopped was called and it is safe to read the number of series.
 	count := series.SeriesCount()
+	series.IterationStopped()
```
Comment (vickenty) [inline pkg/aggregator/demultiplexer.go:621]: It looks that this is always called as soon as `SendIterableSeries` returns. Is there a situation when this wouldn't be true? Would it make sense to move this call into `SendIterableSeries` to make sure it is always called?

File: pkg/aggregator/demultiplexer.go [hunk 3]
```diff
@@ -505,20 +504,17 @@ func (d *AgentDemultiplexer) flushToSerializer(start time.Time, waitForSerialize
 	}
 
 	logPayloads := config.Datadog.GetBool("log_payloads")
-	flushedSeries := make([]metrics.Series, 0)
 	flushedSketches := make([]metrics.SketchSeriesList, 0)
 
 	// only used when we're using flush/serialize in parallel feature
```
Comment (vickenty) [inline pkg/aggregator/demultiplexer.go:509]: Would it make sense to remove this comment now?

File: pkg/aggregator/demultiplexer.go [hunk 4]
```diff
@@ -857,18 +833,15 @@ func (d *ServerlessDemultiplexer) ForceFlushToSerializer(start time.Time, waitFo
 	var seriesSink *metrics.IterableSeries
 	var done chan struct{}
 
-	if d.flushAndSerializeInParallel.Enabled {
-		// only used when we're using flush/serialize in parallel feature
-		logPayloads := config.Datadog.GetBool("log_payloads")
+	// only used when we're using flush/serialize in parallel feature
```
Comment (vickenty) [inline pkg/aggregator/demultiplexer.go:836]: Would it make sense to remove this comment now?

File: pkg/serializer/internal/metrics/service_checks_test.go [hunk 5]
```diff
@@ -151,6 +154,20 @@ func createServiceChecks(numberOfItem int) ServiceChecks {
 	return ServiceChecks(serviceCheckCollections)
 }
 
+func decompressPayload(payload []byte) ([]byte, error) {
```
Comment (vickenty) [inline pkg/serializer/internal/metrics/service_checks_test.go:157]: This appears to be unused?

File: pkg/serializer/internal/metrics/iterable_series.go [hunk 6]
```diff
@@ -47,9 +72,396 @@ func (series IterableSeries) DescribeCurrentItem() string {
...
+type serieIterator interface {
+	MoveNext() bool
+	Current() *metrics.Serie
+}
+
+// MarshalJSON serializes timeseries to JSON so it can be sent to V1 endpoints
+//FIXME(maxime): to be removed when v2 endpoints are available
+func (series IterableSeries) MarshalJSON() ([]byte, error) {
+	// use an alias to avoid infinite recursion while serializing a Series
+	type SeriesAlias Series
```
Comment (vickenty) [inline pkg/serializer/internal/metrics/iterable_series.go:323]: Are we serializing Series to JSON anywhere now? Would it be possible to remove custom serializer and this alias now?

File: pkg/serializer/internal/metrics/iterable_series_test.go [hunk 7]
```diff
@@ -66,34 +63,6 @@ func TestIterableSeriesReceiverStopped(t *testing.T) {
 	iterableSeries.Append(&metrics.Serie{Name: "serie3"})
 }
 
-func TestIterableStreamJSONMarshalerAdapter(t *testing.T) {
```
Comment (vickenty) [inline pkg/serializer/internal/metrics/iterable_series_test.go:69]: Maybe I'm missing something, but why the test is "unused" now? The test subject still seems to be in the code. Or should `NewIterableStreamJSONMarshalerAdapter` be removed too?

**Review by vickenty — COMMENTED** [review]
Mostly good, modulo few nits.

I would also appreciate more detailed QA instructions, especially around OTLP - how to run and how to check that serializer works.

File: pkg/serializer/internal/metrics/iterable_series.go (hunk 1 above)
Comment (ogaca-dd) [reply pkg/serializer/internal/metrics/iterable_series.go]: Unit test added: `TestIterableSeriesEmptyMarshalJSON`

File: pkg/serializer/internal/metrics/iterable_series.go (hunk 6 above)
Comment (ogaca-dd) [reply pkg/serializer/internal/metrics/iterable_series.go:323]: Unfortunately, the Serverless Agent uses this method.

File: pkg/serializer/internal/metrics/service_checks_test.go (hunk 5 above)
Comment (ogaca-dd) [reply pkg/serializer/internal/metrics/service_checks_test.go:157]: This function is used for example at https://github.com/DataDog/datadog-agent/blob/7d83ee116fdc2c94e42b6fb8739eaa585deed1eb/pkg/serializer/internal/metrics/series_test.go#L313.

File: pkg/serializer/internal/metrics/iterable_series_test.go (hunk 7 above)
Comment (ogaca-dd) [reply pkg/serializer/internal/metrics/iterable_series_test.go:69]: This is a good question. 
I wrote initially this test as I wanted to check that `IterableStreamJSONMarshalerAdapter` over a `Series` behaves the same as a `IterableSeries`. As `Series` doesn't implement anymore `StreamJSONMarshaler`, this test must be removed.
As `IterableStreamJSONMarshalerAdapter` is still tested indirectly at several locations (for example in `TestSendV1EventsCreateMarshalersBySourceType`), I did not consider writing a new unit test to only test `IterableStreamJSONMarshalerAdapter`.

**Review by vickenty — APPROVED** [review]

**Pull request merged**